sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -272,18 +272,20 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
-  val HIVE_FILESOURCE_PARTITION_PRUNING =
-    SQLConfigBuilder("spark.sql.hive.filesourcePartitionPruning")
```

Comment:
Is this one public? For public one, should we deprecate it first (not just remove it)?